File: 剑指Offer/变态跳台阶/climbStairsPlus.py
# -*- coding:utf-8 -*-
class Solution:
    def jumpFloorII(self, number):
        # write code here
        if number <2:return number
        res = [0]*(number+1)
        res[0],res[1]=1,1
        for i in range(2,number+1):
            res[i] = sum(res[:i])
        return res[number]

File: 剑指Offer/变态跳台阶/climbStairsPlus.java
public class Solution {
    public int JumpFloorII(int target) {
        if(target<2){
            return target;
        }
        int[] arr = new int[target+1];
        arr[0] = 1;
        arr[1] = 1;
        for(int i = 2; i < target+1; i++){
            for(int j = 0; j <i;j++){
                arr[i]+=arr[j];
            }
        }
        return arr[target];
    }
}